### ISSUE TITLE
egg2dist looks for information in a wrong place

### ISSUE THREAD

**Issue by wvxvw** [pypa/wheel#410]
There are many wrong things happening in `bdsit_wheel`, but this is just too obvious, and hopefully easy to fix.

Here's the problem: I understand that you wanted to find information about the package being packaged, and for some reason decided that the best way to go about it is to call `setup.py install` down the road. Well, `setup.py install` *doesn't have to* generate directories like `./build/wheel/$package_name-$version.egg-info`. It is not necessary for installing packages, nobody needs this directory really, it's purely by chance that such directory ever existed. Just because someone was too lazy to write Zip file directly... oh well.

Anyways. What `egg2dist` should do, is extract this information *from the egg that was built by `setup.py bdist_egg`.* Or, just ask the package maintainer to provide you with this information. Do not assume it will be there, because it won't be. This information *must* be in the egg built using `setuptools`, but it may not be anywhere else.

Now, because of this, I need to patch a bunch of third-party packages, which I will also have to host myself, and will have to keep track of their versions etc. I'm really excited, what a joy!

**Comment by agronholm** [comment]
> There are many wrong things happening in bdsit_wheel

Something that hasn't been reported on this issue tracker?

> Anyways. What egg2dist should do, is extract this information from the egg that was built by setup.py bdist_egg. Or, just ask the package maintainer to provide you with this information. Do not assume it will be there, because it won't be. This information must be in the egg built using setuptools, but it may not be anywhere else.

I expect the `.egg` format, and `bdist_egg` command to eventually go away, as egg is not the canonical binary distribution format of Python (wheel is). Thus relying on `bdist_egg` is just a bad idea.

What is the practical problem you encountered? Your lengthy post does not explain that.

**Comment by wvxvw** [comment]
> Something that hasn't been reported on this issue tracker?

One thing at a time...

> I expect the .egg format, and bdist_egg

As of right now, there's no workable replacement. Wheel is only moderately worse. Conda had an opportunity to come up with a sensible package format, but they didn't even try... What really should die, is the idea that source distribution is an acceptable way to install libraries... but, I'm getting ahead of myself.

Anyways, as of right now, Egg format is a mandatory step in the process of building Wheels, whether you like it or not. Will it go away? -- At the speed this is going, I'll probably retire before it goes away, so, I don't think it makes sense to dismiss it. Things are broken now, and there isn't even a plan for how they are going to work in the future, so, why not fix what we know for sure to be broken?

> What is the practical problem you encountered? Your lengthy post does not explain that.

It's 212 words in 4 paragraphs. An adult without problems with reading comprehension can read this in less than a minute, typically. Granted, English isn't everyone's native language, it should take, maybe a whole of two minutes. Five, if you have coffee breaks in between.

So, what is this shitty attitude? Why? I came to report a problem. Described it in terms even a not very bright person should understand, and you are... just not going to spend a minute reading it? Well, my past interactions with PyPA people say that you won't. Because you seriously couldn't care less. Whoever wrote that code didn't really understand what he or she was doing. They were happy they somehow found *some* way to solve the problem, that would work sometimes. And called it a day.

So, why am I writing this? -- Well, even before I posted this, I knew the problem isn't going to be solved. I just need this sign pole to show others how crappy your tools are so that I would get more freedom to ignore them. That's all.
